Replace static Swagger-ui distribution with unpkg version

theletterf opened the pull request:
This PR removes the static assets for swagger-ui and adds unpkg includes instead. I've also edited the docs accordingly.

- Closes #1520
- Closes #1076

theletterf commented:
@chalin @LisaFC Can't currently test the change. Could you have a look?

LisaFC commented:
Sure, happy to help, looks good from a quick glance but agree we'll need to check it works - is there a reason why you can't test it in your own project?

theletterf commented:
Currently don't have a local Docsy project running and got confused by `docsy-example`. What's the recommended way of testing features?

theletterf commented:
@LisaFC Even though it's slightly out of scope, I've fixed the color issue and added a `_swagger.scss` file which can come in handy for customizing Swagger UI's look and feel a bit (or for hiding elements by using the `display` directive). WDYT?

LisaFC commented:
Oh that looks great! I see what you mean about the background contrast issue - @chalin what do you think?

theletterf commented:
@LisaFC Fixed that in https://github.com/google/docsy/pull/1604/commits/5f31dd2859f1f7f0359daccbf6e1aaf135e3caeb

chalin reviewed (approved):
Great, thanks! LGTM

chalin commented:
@LisaFC - can you give your stamp of approval too?

deining commented on layouts/swagger/baseof.html:
```diff
@@ -22,8 +22,8 @@
             {{ if not .Site.Params.ui.breadcrumb_disable }}{{ partial "breadcrumb.html" . }}{{ end }}
-            <script src="{{ "js/swagger-ui-bundle.js" | relURL }}"></script>
-            <script src="{{ "js/swagger-ui-standalone-preset.js" | relURL }}"></script>
+            <script src="https://unpkg.com/swagger-ui-dist@5.1.0/swagger-ui-bundle.js" crossorigin></script>
```
I would propose to add the SRI hash here (as we do with other scripts).
Same for other files pulled in from `unpkg.com`.

chalin replied:
That works for me.

theletterf replied:
Edited to add the SRI hash. Thanks!

TIL that you can append `?meta` to any package in unpkg to retrieve package metadata, including SRI hashes: https://github.com/mjackson/unpkg/issues/48#issuecomment-323161338. That, or using srihash.org.

deining replied:
> Edited to add the SRI hash. Thanks!
> 
> TIL that you can append `?meta` to any package in unpkg to retrieve package metadata, including SRI hashes: [mjackson/unpkg#48 (comment)](https://github.com/mjackson/unpkg/issues/48#issuecomment-323161338). That, or using srihash.org.

Thanks for the info. Up to now, I didn't know that either.

deining reviewed (approved):
LGTM now.
Can you rebase your PR to HEAD of main, please?

theletterf commented:
@deining Done

theletterf commented:
Can't merge on my own!

chalin commented:
@LisaFC - still awaiting your approval here

LisaFC reviewed (approved):
Looks fantastic - sorry for the late review.
